src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -46,13 +47,110 @@ constexpr std::wstring_view StateCollapsed{ L"Collapsed" };
 DEFINE_ENUM_FLAG_OPERATORS(winrt::Microsoft::Terminal::Control::CopyFormat);
 DEFINE_ENUM_FLAG_OPERATORS(winrt::Microsoft::Terminal::Control::MouseButtonState);
 
+// WinUI 3's UIElement.ProtectedCursor property allows someone to set the cursor on a per-element basis.
+// This would allow us to hide the cursor when the TermControl has input focus and someone starts typing.
+// Unfortunately, no equivalent exists for WinUI 2 so we fake it with the CoreWindow.
+// There are 3 downsides:
+// * SetPointerCapture() is global state and may interfere with other components.
+// * You can't start dragging the cursor (for text selection) while it's still hidden.
+// * The CoreWindow covers the union of all window rectangles, so the cursor is hidden even if it's outside
+//   the current foreground window, but still on top of another Terminal window in the background.
+static void hideCursorUntilMoved()
+{
+    static CoreCursor previousCursor{ nullptr };
+    static auto shouldVanish = []() {

```

Comment:
And hopefully correct! I don't 100% trust the WinUI mouse tracking API because in my testing it would randomly enter a state where it stops signaling the callbacks. The problem is that I never figured out if that was due to my incorrect input panel code (also in this file) or due to a WinUI bug, but the former was the only way I could make it fail reliably.
If we end up having issues with this code I'll commit a workaround that tracks the cursor via our standard HWND message handlers.